### ISSUE TITLE
Target zones not displaying on peloton power zone ride

### ISSUE THREAD

**Issue by Debbles12** [cagnulein/qdomyos-zwift#744]
No target zones during ride.  Did see name of peloton ride in qz app.

[debug-Mon_Mar_28_05_55_02_2022.log](https://github.com/cagnulein/qdomyos-zwift/files/8364901/debug-Mon_Mar_28_05_55_02_2022.log)

**To Reproduce**
1. Start qz app
2. start Peloton app.
3. start power zone ride.
4. hit start button on qz app.

**Expected behavior**
Should see target zones throughout ride.

**Screenshots**

**Smartphone (please complete the following information):**
 - Device: Android Samsung galaxy tab s6 lite
 - OS: 
 - Version Android 11

**Append a debug log**

 Follow this guide https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F
  
  
**Additional context**

**Comment by SayrahD** [comment]
Same bug from Christine, this one with PZP credentials; next comment will be with credentials removed and no popup from HFB
[debug-Mon_Mar_28_10_06_46_2022.log](https://github.com/cagnulein/qdomyos-zwift/files/8365363/debug-Mon_Mar_28_10_06_46_2022.log)

**Comment by SayrahD** [comment]
PZP credentials same bug.  
[debug-Mon_Mar_28_10_04_01_2022.log](https://github.com/cagnulein/qdomyos-zwift/files/8365372/debug-Mon_Mar_28_10_04_01_2022.log)

**Comment by cagnulein** [comment]
@SayrahD both log have the pzp username. maybe wrong log?
anyway @SayrahD @Debbles12 i would like to double check if the same issue persists in the latest beta version. Can I send you it? In case send me an email to roberto.viola83@gmail.com indicating this ticket and if you're on ios or android
Thanks!

**Comment by SayrahD** [comment]
@cagnulein  here you go!   
[debug-Mon_Mar_28_21_46_21_2022.log](https://github.com/cagnulein/qdomyos-zwift/files/8367699/debug-Mon_Mar_28_21_46_21_2022.log)
Name of ride populates immediately, but no resistance follow pop up window.  Thanks!

**Comment by mnasman1** [comment]
[debug-Tue_Mar_29_06_18_53_2022.log](https://github.com/cagnulein/qdomyos-zwift/files/8371244/debug-Tue_Mar_29_06_18_53_2022.log)
Here is mine as well.

**Comment by mnasman1** [comment]
I dont have a hfb account but I do have a pzp login so thats what I have been using.

**Comment by cagnulein** [comment]
@mnasman1 hfb doesn't require an account, QZ use HFB if it found "username" in the PZP username field.

**Comment by cagnulein** [comment]
@SayrahD @Debbles12 @mnasman1 i just received the confirmation from Christine that HFB works well. So until Angie will resolve this issue, i suggest to put "username" to PZP username

**Comment by mnasman1** [comment]
Ok do you think this was intentional by Angie? Also for hfb I assume its username without quotes? Blanks password?

**Comment by cagnulein** [comment]
@mnasman1 no it wasn't intentional, she told me that they did some changes but they didn't think to break something. They just not add QZ as a test bench.

yes username without quotes, password could be whatever you want

**Comment by Jedimama** [comment]
[debug-Sun_Apr_3_11_15_01_2022.log](https://github.com/cagnulein/qdomyos-zwift/files/8405099/debug-Sun_Apr_3_11_15_01_2022.log)

Same issue but I’ve only ever used HFB & never had a PZP username. It’s only a problem with power zone rides & has happened twice now.

**Comment by cagnulein** [comment]
hi @Jedimama thanks for reporting this! I checked your log and the issue is that workout there isn't on hfb. Your workout is from  04/23/21 by Christine D'Ercole but on HFB there is the workout for that date
<img width="1627" alt="Screenshot 2022-04-03 at 20 20 29" src="https://user-images.githubusercontent.com/48216/161442364-83f0e4cb-a04c-42da-b4af-8bdbb3881340.png">

could you please try with another ride?

**Comment by Jedimama** [comment]
> hi @Jedimama thanks for reporting this! I checked your log and the issue is that workout there isn't on hfb. Your workout is from 04/23/21 by Christine D'Ercole but on HFB there is the workout for that date <img alt="Screenshot 2022-04-03 at 20 20 29" width="1627" src="https://user-images.githubusercontent.com/48216/161442364-83f0e4cb-a04c-42da-b4af-8bdbb3881340.png">
> 
> could you please try with another ride?

Weird! I’m doing the Build Your Power Zones program and all 3 of the first weeks classes have that date & time. The first one is Matt Wilpers, which worked fine. The second two are different teachers & neither one worked! Maybe HFB ignored the second two workouts thinking it was all the same class because they all had the same date & time?

**Comment by cagnulein** [comment]
@Jedimama are all there in this log? if so i will check them tomorrow (11pm here)

**Comment by Jedimama** [comment]
> @Jedimama are all there in this log? if so i will check them tomorrow (11pm here)

No, they’re not. But I checked the peloton buddy site for the rest of the classes in this program & then I checked the HFB site & it looks like I’m going to keep having this issue. I emailed HFB. Thanks for your help!

**Comment by cagnulein** [comment]
@Jedimama hfb is not the issue, please share a log with a different class so i can check it. Thanks

**Comment by cagnulein** [comment]
[debug-Mon_Apr_4_12_36_52_2022.log](https://github.com/cagnulein/qdomyos-zwift/files/8412410/debug-Mon_Apr_4_12_36_52_2022.log)
Christine log with highlighting that the issue is on the AuthenticatedApi_GetClassByPeletonId api that doesn't answer
Already wrote again to Angie

**Comment by Jedimama** [comment]
> @Jedimama hfb is not the issue, please share a log with a different class so i can check it. Thanks


Same issue today. Ignore my ridiculous heart rate. Either my monitor is having a breakdown or I’m having a heart attack. 

[debug-Tue_Apr_5_08_46_14_2022.log](https://github.com/cagnulein/qdomyos-zwift/files/8418574/debug-Tue_Apr_5_08_46_14_2022.log)

**Comment by cagnulein** [comment]
@Jedimama 
<img width="1637" alt="Screenshot 2022-04-05 at 19 00 50" src="https://user-images.githubusercontent.com/48216/161809705-6d629c2b-a772-4aea-9638-32668e1a060c.png">
same issue, the ride from Denis there isn't in HFB. I don't know what to say, if HFB hasn't it, QZ can't get the metrics :(

**Comment by Jedimama** [comment]
Yup. That’s why I think it’s an HFB issue & not QZ. I appreciate you helping me figure that out.

**Comment by xpmurph48** [comment]
I am also not seeing the power zone targets or auto resistance is no longer working. Attaching the logs from ride this morning. 

[debug-Mon_Apr_11_08_09_57_2022.log](https://github.com/cagnulein/qdomyos-zwift/files/8465609/debug-Mon_Apr_11_08_09_57_2022.log)

**Comment by xpmurph48** [comment]
> Are you using PZP or HFB? I have not had any issues since switching to HFB. There is currently an issue with PZP's api.

Yes, I have been using HFB, in the app I just have USERNAME and the name and no password.  This is based on feedback from Robert.  Still not working and he asked me to upload the debug logs.

**Comment by cagnulein** [comment]
@xpmurph48 the issue is the same of @janabouc 's :) username must be all lowercase, in your case is Username. Just put "username" (without quotes) and press ok. It will work ;)

**Comment by xpmurph48** [comment]
> @xpmurph48 the issue is the same of @janabouc 's :) username must be all lowercase, in your case is Username. Just put "username" (without quotes) and press ok. It will work ;)

Thanks for catching that...glad it was an easy one :)

**Comment by cagnulein** [comment]
[debug-Fri_Apr_15_11_21_48_2022.log](https://github.com/cagnulein/qdomyos-zwift/files/8497138/debug-Fri_Apr_15_11_21_48_2022.log)

Christine sends me this file: in this Peloton doesn't send the title name of the workout, very strange
